plugins/vpc-bridge/config/k8s_connector.go

```diff
@@ -0,0 +1,209 @@
+	"github.com/Microsoft/go-winio/pkg/guid"
+	log "github.com/cihub/seelog"
+)
+
+// Represents output of k8s connector received from named pipe
+type k8sExecutionResult struct {
+	stdout string // Represents message with required data.
+	stderr error  // Represents error while getting output.
+}
+
+const (
+	// Represents timeout for reading over results channel.
+	// Represents timeout for reading from connection object.
+	k8sExecutionTimeout = 3 * time.Minute
+)
+
+func init() {
+	retrievePodConfigHandler = retrievePodConfig
+}
+
+// retrievePodConfig retrieves a pod's configuration from an external source.
+func retrievePodConfig(netConfig *NetConfig) error {
+
+	kc := netConfig.Kubernetes
+
+	// retrieve Pod IP Address CIDR string using podName, podNamespace.
+	ipAddress, err := retrievePodIPWithK8sConnector(kc.Namespace, kc.PodName)
+	if err != nil {
+		return fmt.Errorf("failed to get pod IP %s: %w", kc.PodName, err)
+	}
+
+	// Parse IP address returned by k8s binary.
+	ipAddr, err := vpc.GetIPAddressFromString(ipAddress)
+	if err != nil {
+		return fmt.Errorf("invalid IPAddress %s from pod label", ipAddress)
+	}
+	netConfig.IPAddresses = append(netConfig.IPAddresses, *ipAddr)
+
+	return nil
+}
+
+// retrievePodIPWithK8sConnector retrieves Pod IP with k8s connector binary using named pipe.
+func retrievePodIPWithK8sConnector(podNamespace, podName string) (string, error) {
+
+	// Create named pipe, used to get output from k8s connector binary.
+	// pipeListener will accept connections in separate go routine. Output of binary is received via channel.
+	pipeListener, executionResultChan, err := createNamePipeForOutputFromK8sConnector()
+	if err != nil {
+		return "", fmt.Errorf("retrievePodIPWithK8sConnector: error creating named pipe %w", err)
+	}
+	defer pipeListener.Close()       // Defer closing named pipe listener.
+	defer close(executionResultChan) // Defer closing result channel.
+
+	namedPipePath := getPipeName(pipeListener)
+	log.Debugf("named piped: %s created", namedPipePath)
+
+	// Execute k8s connector binary with pod namespace, pod name and named pipe path.
+	err = executeK8sConnector(podNamespace, podName, namedPipePath)
+	if err != nil {
+		return "", fmt.Errorf("retrievePodIPWithK8sConnector: error executing k8s connector %w", err)
+	}
+
+	// Read output of k8s binary from named pipe.
+	var result k8sExecutionResult
+	select {
+	case result = <-executionResultChan:
+	case <-time.After(k8sExecutionTimeout):
+		return "", fmt.Errorf("retrievePodIPWithK8sConnector: timeout getting output of k8s connector")
+	}
+
+	if result.stderr != nil {
+		return "", fmt.Errorf("retrievePodIPWithK8sConnector: error in k8s connector %w", result.stderr)
+	}
+	podIP := result.stdout
+	log.Debugf("got pod IP: %s for pod %s in namespace %s", podIP, podName, podNamespace)
+
+	return podIP, nil
+}
+
+// executeK8sConnector executes aws-vpc-cni-k8s-connector binary and get pod IP over named pipe.
+// Execution logs from binary are stderr.
```

Comment:
Execution logs from binary are stdout and stderr.

Reply:
By default, logrus throws all the logs to `stderr`

Reply:
changed it to `stdout` and will change binary too to throw logs to `stdout`